gptme/tools/subagent.py

```diff
@@ -41,23 +49,49 @@ class ReturnType:
     result: str | None = None
 
 
-@dataclass(frozen=True)
+@dataclass
```

Comment:
I don't think you should remove the frozen=True. Find some other way to communicate (attach a mutable ref or something) instead of mutating the object.